### PR TITLE
Enhancement/highcharts disable diagramtypes cleanup

### LINKED ISSUE #760: FHI Angular Highcharts: rydd opp i "disable tests"
### Description

Over tid har testene for å sjekke om en diagramtype kan vises eller ikke blitt veldig rotete. Noen er ikke lenger i bruk, noen mangler, og koden er rotete.

### Acceptance Criteria

1. Gå til https://nice-bush-[ID-kommer].westeurope.2.azurestaticapps.net/developer/components/highcharts
2. Klikke igjennom alle eksempler, og se i konsollen at warnings for diagramtyper som ikke kan vises fortsatt gir mening.

### PR DESCRIPTION
**Pull request by proand**
Remove two redundant disable tests, add two new, and refactor all disable tests.

Remove:

- `noSeriesOrNoData`
- `allDataInOneOrMoreSeriesAreFlagged`

Add:

- `notAllUnitsFoundInSeries`
- `onlyOneSerieAndAllDataAreFlagged`

Refactoring:

- Add enum msgId
- Remove diagram type switch, use if tests instead
  - Test for one or more diagram type pr. test
  - Call one or more test inside the if tests
- Rename, and rewrite all the disable tests

This closes #760

### CODE REVIEW

**Comment by proand** [comment]
@VillereV En liten heads up: denne PR-en inneholder også testen `notMaxOneUnitsInSeries`, så den gjør strengt tatt PR #756 redundant.

File: projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts [hunk 1]
```diff
@@ -13,21 +12,33 @@ import { DiagramType } from '../models/diagram-type.model';
...
   private diagramTypeDisabledWarnings: { [key in FhiDiagramTypeIds]?: string } = {};
-  private diagramTypeDisabledWarningsText = {
-    flaggedData: 'series.length > 1 && flaggedSeries?.length !== 0',
-    moreThanOneSeries: 'series.length > 1',
-    notTwoUnits: 'diagramOptions.units?.length !== 2',
-    notMinTwoUnitsInSeries: 'this.uniqueUnitIdCountInSeries() < 2',
-    notGeo: 'series.length === 1 && isNotGeo(this.series[0])',
-    noSeriesOrNoData: 'this.series.length === 0 || this.series[0].data.length === 0',
-    allDataInOneOrMoreSeriesAreFlagged: 'allDataInOneOrMoreSeriesAreFlagged',
+  private diagramTypeDisabledWarningMessages = {
```
Comment (VillereV) [inline projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts:33]: Could `diagramTypeDisabledWarningMessages` be typed as `Record<msgId, string>`?

File: projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts [hunk 2]
```diff
@@ -171,123 +176,144 @@ export class DiagramTypeGroupService {
...
+      this.updateDisabledWarnings(diagramType.id, msgId.moreThanOneSeries);
       return true;
     }
     return false;
   }
 
-  private disableColumn(): boolean {
-    if (this.disableBar()) {
-      this.diagramTypeDisabledWarnings.column = this.diagramTypeDisabledWarningsText.flaggedData;
+  private notAllUnitsFoundInSeries(diagramType: DiagramType): boolean {
```
Comment (VillereV) [inline projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts:252]: `allUnitsFoundInSeries` is hard to read. Simplify to  `const allUnitsFoundInSeries = this.diagramOptions.units.every((unit) =>
      this.series.some((serie) => serie.unitId === unit.id)
    );`?

Comment (proand) [reply projects/fhi-angular-highcharts/src/lib/services/diagram-type-group.service.ts:252]: Nice! Much better 🙂

**Pull request merged**